source-hubspot-native/source_hubspot_native/api.py

```diff
@@ -271,14 +460,260 @@ async def fetch_recent_tickets(
     log: Logger, http: HTTPSession, since: datetime, cursor: PageCursor
 ) -> tuple[Iterable[tuple[datetime, str]], PageCursor]:
 
-    url = f"{HUB}/crm-objects/v1/change-log/tickets"
-    params = {"timestamp": int(since.timestamp() * 1000) - 1}
+    url = f"{HUB}/crm/v3/objects/tickets"
+    
+    result = OldRecentTicket.model_validate_json(
+        await http.request(log, url, params=None)
+    )
+
+    return (
+        (r.updatedAt, str(r.id))
+        for r in result.results
+    ), None and None
+
+async def fetch_recent_contacts_lists(
+    log: Logger, http: HTTPSession, since: datetime, page: PageCursor
+) -> tuple[Iterable[tuple[datetime, str, Any]], PageCursor]:
+
+    url = f"{HUB}/contacts/v1/lists"
+    params = {"count": 100, "offset": page} if page else {"count": 250}
+
+    data = await http.request(log, url, params=params)
+
+
+    result = V1CustomCRMObject2.model_validate_json(
+        data
+    )
+
+    for item in result.lists:
+        item['listId'] = str(item['listId'])
+
+    if result.hasMore is False:
```

Comment:
Other similar seeming functions use `result.has_more and result.time_offset` as the return value for the `PageCursor`, seemingly as an alternative for this if/else. Would that work for these too?

Reply:
They are being handled by the fetch_changes_no_batch function, more on how it works in their docstring added here a484280098b57da301ffc5bc09ab89fa835fc8d0